### DISCUSSION TITLE
How to stop CarlaUE4 window to stop lauching itself?

### DISCUSSION THREAD

**Discussion by i-kj** [guardstrikelab/carla_apollo_bridge#132] [Q&A] [answered]
Hello All,
Since the time I installed Carla apollo bridge, the CarlaUE4 application is auto launching itself despite every time I tried closing it. 

I performed the below steps so far to troubleshoot the issue:

1. Identify the process and kill: 

_smit@smit-BUSINESSline:~/CARLA_0.9.13/PythonAPI/examples$ pgrep CarlaUE4
52540
52577_

_smit@smit-BUSINESSline:~/CARLA_0.9.13/PythonAPI/examples$ kill 52540 52577_

Application auto launched as usual.

2. Checked startup Applications and CarlaUE4 is not in the list and also cannot find any autostart folder _cd ~/.config/autostart_

3. Deleted _restart always_ from Carla-compose.yml file but still the issue keeps happening.

Some additional information that might help:

_smit@smit-BUSINESSline:~$ ps aux | grep Carla
smit       53644  0.0  0.0   4644   808 ?        Ss   13:48   0:00 /bin/sh ./CarlaUE4.sh -quality-level=Low -carla-rpc-port=2000
smit       53681  334  7.7 4880416 2525556 ?     Rl   13:48  41:54 /home/carla/CarlaUE4/Binaries/Linux/CarlaUE4-Linux-Shipping CarlaUE4 -quality-level=Low -carla-rpc-port=2000
smit       54246  0.0  0.0  10000   656 pts/0    S+   14:01   0:00 grep --color=auto Carla_

_smit@smit-BUSINESSline:~$ docker ps
CONTAINER ID   IMAGE                                              COMMAND                  CREATED      STATUS         PORTS     NAMES
444e13091c3c   apolloauto/apollo:dev-x86_64-18.04-20210914_1336   "/bin/bash"              4 days ago   Up 2 minutes             apollo_dev_smit
0c05275b9284   carlasim/carla:0.9.14                              "./CarlaUE4.sh -qual…"   5 days ago   Up 2 hours               carla-simulator-1_

docker stop 0c05275b9284 is stopping the simulator from launching again but when I restart the system the carla simulator docker is active back again. Is this the way it works? 

I am ready to provide additional information if needed. Any help would be really appreciated. 

Thanks in advance.

**Comment by XiaoFei9704** [comment, marked as answer]
As I mentioned earlier：

> See the last line of scripts/carla-compose.yml:
>        restart: always
> If you do not want Carla to auto-restart, just delete it

Just doing this is not enough because the Docker container has already been created, and the current container will still automatically restart.
So you should first delete the carla container and then rerun 'docker_run_carla. sh'
`
docker rm -f carla-simulator-1

./docker_run_carla.sh
`
